client/src/components/modal-forms/ManageGraduationForm.js

```diff
@@ -21,12 +21,13 @@ export const ManageGraduationForm = ({ initialValues, onClose, onSubmit, cohortE
         onSubmit={onSubmit}
         validationSchema={ParticipantPostHireStatusSchema}
       >
-        {({ submitForm, values, setFieldValue }) => {
+        {({ submitForm, values, setFieldValue, setFieldTouched }) => {
           const handleStatusChange = ({ target }) => {
             const { value } = target;
             setFieldValue('status', value);
             if (value === postHireStatuses.postSecondaryEducationCompleted) {
               setFieldValue('data.date', cohortEndDate);
+              setFieldTouched('data.date', false);
```

Comment:
Wouldn't the issue be that the field is failing validation rather than if it's touched or not?

Reply:
I wonder if using the shouldRevalidate option would solve this
```
              setFieldValue('data.date', cohortEndDate, true);
```
https://formik.org/docs/api/formik#setfieldvalue-field-string-value-any-shouldvalidate-boolean--void

Reply:
Do we know where that delay in the video comes from? It looks like the warning message only lasts for a second or so

Reply:
When I was interacting with it, the error stayed up until I clicked the form. That's why I went with the "touched" route.

In my head it makes more sense that the touched value should be TRUE, not FALSE, which is the opposite of what I did. I'm thinking that due to the input's schema changing depending on the value of another input, these events fire in the following order and that's the root cause:
- change the status from Unsuccessful to Graduated
- handleStatusChange updates the graduated date
- the date field registers as touched
- the error message is displayed
- the schema changes to the graduated schema